### ISSUE TITLE
Adjusting map sizes to avoid black zoom, add map size enormous

### ISSUE THREAD

**Issue by ArchdukeLiet** [We-the-People-civ4col-mod/Mod#723]
Hi everyone,

This week I've been looking into map sizes and investigated in more detail when black zoom appears. Black zoom is when part of the world map is black when you zoom out at the world map. As Nightinggale discovered earlier, black zoom is present depending on a X and Y ratio.

I've conducted 25 tests*:
![image](https://user-images.githubusercontent.com/116034090/215865074-05cd88a9-68a2-45df-a0c0-97c11b9037b0.png)

The proposal for adjustments would be twofold:
1. Adjust gigantic and huge map sizes.
2. Add an intermediate map, 'enormous'.

On (1), to avoid black zoom on a gigantic map (now 108x215, in total 23220 tiles), the map ratio would need to be adjusted. The most closely related map size without black zoom would be 127x184, in total 23368 tiles. With this ratio it is aimed to maximise Y, as a maximum Y would allow the most space for European nations to settle on (i.e. longest East coast).
The huge map currently is working fine, but could potentially have black zoom as FaireWeatherTweakEx makes some minor adjustments to the map size. To avoid this, in FaireWeatherTweakEx the world size should be fixed to 58x100 instead of depending on mc.distanceToEurope.

On (2), there is a big step from huge (5800 tiles) to gigantic (23220 tiles). It will likely be facilitating to a number of players to have an intermediate map size. This map size, let's call it 'Enormous', could be 85x137 (11645 tiles). This ratio at this size would not lead to black zoom appearing. 
With the enourmous map size the aim is to introduce a map size that does include all natives and European powers, while allowing more of the new world to be colonised by a player (keeping the number of player settlements more equal. At least in my experience I often end up without using large parts of gigantic maps.

Of course I'm curious about the opinions of other modders on this. And of course if someone could commit to making the necessary changes.

----------------
More detailed findings and conclusions, copied from Discord:
- Indeed 1.45 seems to be a good number to divide Y with at 'gigantic' map sizes (i.e. Y is 215).
- This however does mean that also a gigantic map size would increase from 23220 to 31820 tiles (+37%).
- With this we might need to find a new size that is more or less similar to 23220 tiles, however changing the map ratio impacts gameplay as Europeans will be closer to each other. 
- Curiously enough the 1.45 does not hold in all cases: a huge map has no black zoom with 1.72. 
- So the ratio is not a constant factor but changes depending on the size of Y.
- Looking for a map size between huge and gigantic we would aim at a map size of about 11600+ tiles.
- Preferably, the ratio should be as high as possible, as it stretches the east coast more and increases distances between Europeans.
- I ended up with 85x137 (11645 tiles), ratio 1.61, to have no black zoom.
- My proposal is to include a new map size, Enormous, to have this size.
- What is interesting is that doubling the X and Y of the huge map (58 to 116 and 100 to 200) does not work for a gigantic map. This means that the map size influences the max divide factor. Apparently for about the gigantic size the magical number (as discovered previously by Nightinggale) is 1.45.
- My proposal is to adjust the gigantic map size to 127x184.

**Comment by Nightinggale** [comment]
I agree with the proposed solutions. Having random maps not turn black would be an improvement.
What's not clear here is that FaireWeatherTweakEx adds extra width based on the min distance between land and Europe access plots. I propose disabling this feature for the biggest maps as it only really makes a difference on small maps and we will gain better control of the map size if it's fixed regardless of map options.

Also this is related to issues #350 and #353. Fixing this should close all 3 issues.

**Comment by XSamatan** [comment]
As I'm one of the players enjoying the longer travel times to the ocean plots in FaireWeatherTweakEx, I would be saddened to see this feature go. Furthermore, I hardly ever encounter the described black zoom, however if this is an greater issue for other players a fix would be surely needed.

Regards
XSamatan

**Comment by Nightinggale** [comment]
> As I'm one of the players enjoying the longer travel times to the ocean plots in FaireWeatherTweakEx

The proposal here isn't to remove the ability to increase travel time, but rather to support it without adding extra width to the map. In other words 127x184 would with the current setup has 127 width with 4 plot distance and 129 with 6 plot distance. Since we have to add extra width anyway to get around the black terrain issue, the proposal is to have a width of 127 regardless of travel distance. The distance between Europe access plots and land will still differ depending on map settings. No change there.

Part of the reason to use fixed width is that it would require us to test multiple map sizes for the black terrain issue. Once the width is more than 100 anyway, one more or less won't make a lot of difference and it greatly reduce the number of map variation we would need to test.
